src/native-common/WebView.tsx

```diff
@@ -24,27 +33,53 @@ const _styles = {
 type MixedContentMode = 'never' | 'always' | 'compatibility' | undefined;
 
 export class WebView extends React.Component<RX.Types.WebViewProps, RX.Types.Stateless> implements RX.WebView {
-    private _mountedComponent: RN.WebView | undefined;
+    private _mountedComponent: RNWebView | undefined;
 
     render() {
         const styles = [_styles.webViewDefault, this.props.style] as RN.StyleProp<RN.ViewStyle>;
         const source = this._buildSource();
 
         // Force use of webkit on iOS (applies to RN 0.57 and newer only).
-        const extendedProps: RN.ExtendedWebViewProps = {
+        const extendedProps: RNWebViewProps = {
             useWebKit: true
         };
 
+        // Keep compatibility with old code that uses window.postMessage. For more information,
+        // see https://github.com/react-native-community/react-native-webview/releases/tag/v5.0.0
+        const injectedJavascript = `
+        // WebView -> Native
+        (function() {
+            if (window.ReactNativeWebView && window.ReactNativeWebView.postMessage) {
+                window.postMessage = function(data) {
+                    window.ReactNativeWebView.postMessage(data);
+                };
+            }
+        })();
+
+        // Native -> WebView
+        (function() {
+            window.postMessageFromReactXP = function(message) {
+                var event;
+                try {
+                    event = new MessageEvent('message', { data: message });
+                } catch (e) {
+                    event = document.createEvent('MessageEvent');
+                    event.initMessageEvent('message', true, true, data.data, data.origin, data.lastEventId, data.source);
+                }
+                document.dispatchEvent(event);
+            };
+        })();`;
```

Comment:
The react-native-webview guide suggests appending ```true;``` to the end of your injected script to avoid 'silent failures'. I wasn't able to find any more details, but may want to follow their guidance
https://github.com/react-native-community/react-native-webview/blob/master/docs/Guide.md#communicating-between-js-and-native